src/org/openstreetmap/josm/data/validation/OsmValidator.java

```diff
@@ -113,49 +114,50 @@ private OsmValidator() {
+        CrossingWays.SelfCrossing.class,                    // ID   601 ..   699
+        SimilarNamedWays.class,                             // ID   701 ..   799
+        Coastlines.class,                                   // ID   901 ..   999
+        WronglyOrderedWays.class,                           // ID  1001 ..  1099
+        UnclosedWays.class,                                 // ID  1101 ..  1199
+        TagChecker.class,                                   // ID  1201 ..  1299
+        UnconnectedWays.UnconnectedHighways.class,          // ID  1301 ..  1399
+        UnconnectedWays.UnconnectedRailways.class,          // ID  1301 ..  1399
+        UnconnectedWays.UnconnectedWaterways.class,         // ID  1301 ..  1399
+        UnconnectedWays.UnconnectedNaturalOrLanduse.class,  // ID  1301 ..  1399
+        UnconnectedWays.UnconnectedPower.class,             // ID  1301 ..  1399
+        DuplicateWay.class,                                 // ID  1401 ..  1499
+        NameMismatch.class,                                 // ID  1501 ..  1599
+        MultipolygonTest.class,                             // ID  1601 ..  1699
+        RelationChecker.class,                              // ID  1701 ..  1799
+        TurnrestrictionTest.class,                          // ID  1801 ..  1899
+        DuplicateRelation.class,                            // ID  1901 ..  1999
+        WayConnectedToArea.class,                           // ID  2301 ..  2399
+        PowerLines.class,                                   // ID  2501 ..  2599
+        Addresses.class,                                    // ID  2601 ..  2699
+        Highways.class,                                     // ID  2701 ..  2799
+        BarriersEntrances.class,                            // ID  2801 ..  2899
+        OpeningHourTest.class,                              // ID  2901 ..  2999
+        MapCSSTagChecker.class,                             // ID  3000 ..  3099
+        Lanes.class,                                        // ID  3100 ..  3199
+        ConditionalKeys.class,                              // ID  3200 ..  3299
+        InternetTags.class,                                 // ID  3300 ..  3399
+        ApiCapabilitiesTest.class,                          // ID  3400 ..  3499
+        LongSegment.class,                                  // ID  3500 ..  3599
+        PublicTransportRouteTest.class,                     // ID  3600 ..  3699
+        // ID  3700 ..  3799 is automatically removed since it clashed with pt_assistant.
+        SharpAngles.class,                                  // ID  3800 ..  3899
+        ConnectivityRelations.class,                        // ID  3900 ..  3999
+        DirectionNodes.class,                               // ID  4000 ..  4099
+        RightAngleBuildingTest.class,                       // ID  4100 ..  4199
+        CycleDetector.class,                                // ID  4200 ..  4299
```

Comment:
The IDs are no longer as important to document, since we (effectively) randomly generate a base number for each test class. Still not a bad idea to document them though.

Reply:
Alright, I can remove these changes. I don't like it too much either.

Reply:
Reverted the change.